Only print NUMA nodes if requested

bwatkinson opened the pull request:
After updating the build system to CMake XDD was outputing the NUMA domain for each worker thread in the header of the output. This address that issue by printing that info in the event it is requested through the `-debug` flag.

Since the `GO_DEBUG_USER1` global debug was never used, it was updated to be used for init debugging and changed to GO_DEBUG_INIT. So if the user does `-debug init` they will now see what cpus the worker threads have been pinned to.

As part of the work, I also reorganized the test directory a bit. I made a separate debug test directory just to test out this functionality as it really did not make sense for it to be in the fuctional test cases. I placed all the common test CMake variables in the CMakeLists.txt and then just added individual test information in the functional and debug CMakeLists.txt files.

Fixes #13
Signed-off-by: Brian Atkinson <batkinson@lanl.gov>

bwatkinson committed:
Only print NUMA nodes if requested

After updating the build system to CMake XDD was outputing the NUMA
domain for each worker thread in the header of the output. This address
that issue by printing that info in the event it is requested through
the `-debug` flag.

Since the `GO_DEBUG_USER1` global debug was never used, it was updated
to be used for init debugging and changed to GO_DEBUG_INIT. So if the
user does `-debug init` they will now see what cpus the worker threads
have been pinned to.

As part of the work, I also reorganized the test directory a bit. I made
a separate debug test directory just to test out this functionality as
it really did not make sense for it to be in the fuctional test cases. I
placed all the common test CMake variables in the CMakeLists.txt and
then just added individual test information in the functional and debug
CMakeLists.txt files.

Fixes #13
Signed-off-by: Brian Atkinson <batkinson@lanl.gov>

calccrypto commented on tests/debug/CMakeLists.txt:
```diff
@@ -0,0 +1,44 @@
+    test_xdd_debug_init.sh
+    )
+
+# original tests depended on xdd being installed
+option(TEST_INSTALLED "Use installed xdd for tests" ON)
+if (TEST_INSTALLED)
+  set(XDD_EXEC "$<TARGET_FILE:xdd>")
+else()
+  set(XDD_EXEC "${CMAKE_BINARY_DIR}/bin/xdd")
+endif()
```
Duplicate

bwatkinson replied:
Ughhh... I forgot to cleanup some stuff before creating the PR. I just put another commit on top of the original that cleans up everything. We need to squash this down into the original commit if we decide to merge this.

bwatkinson committed:
Cleaning up based on PR feedback

SQAUSH before merging

I overlooked cleaning up the CMakeLists.txt in the debug test case
directory. This fixes that oversight.

Signed-off-by: Brian Atkinson <batkinson@lanl.gov>

calccrypto commented on tests/debug/CMakeLists.txt:
```diff
@@ -0,0 +1,12 @@
+foreach(TEST ${DEBUG_TESTS})
+  configure_file("${TEST}" "${TEST}" COPYONLY)
+  add_test(NAME "${TEST}"
+    COMMAND sh -c "${CMAKE_CURRENT_BINARY_DIR}/${TEST} 2> ${TESTDIR}/logs/${TEST}.log"
```
Why use `sh`? The script file should just be made executable

bwatkinson replied:
No idea, that is what we originally had before lol. Let me try seeing if modifying it works.

calccrypto replied:
If the goal is to redirect the output, the redirection can happen in the test instead

bwatkinson replied:
Again, this is what originally was there. I didn't add that lol... I am testing it out now to see what the point of it was.

bwatkinson replied:
Honestly, I don't think we even need to have the log stuff at all... I am super confused on what the goal there was. Maybe the idea was do `-x` or something? Like to print out the statements that were running? Who put that `sh -c` in there originally?

bwatkinson replied:
I just removed the redirection to the log thing completely. See my latest commit comment. I seriously have no idea what the intention was there, so I just got rid of it.

calccrypto commented on tests/debug/test_xdd_debug_init.sh:
```diff
@@ -0,0 +1,36 @@
+
+echo "Foobar and I am here motherfucker"
+
+xdd_cmd="${XDDTEST_XDD_EXE} -op write -reqsize 128 -numreqs 1 -targets 1 /dev/null -verbose -debug INIT"
+${xdd_cmd} 2>&1 | grep "bound to NUMA node"
```
Putting the command into a string only to run it in the next line seems superfluous. It also opens up the command for bad strings in `XDDTEST_XDD_EXE`

bwatkinson replied:
Updated.

bwatkinson committed:
More updates based on PR feedback

SQUASH before merging

Made updates to test_xdd_debug_init.sh to just call XDD directly without
putting it in a string first.

Not sure what the idea was behind the redirection of the shell scripts
to the log file was. As fas as I can tell, it never resulted in anything
being written out to test-dir/logs. For now, I am just getting rid of
the redirection till someone can show me what it was suppose to do.

Signed-off-by: Brian Atkinson <batkinson@lanl.gov>
